internal/services/appservice/static_web_app_link_backend.go

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package appservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/apimanagementservice"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerapps/2023-05-01/containerapps"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/web/2023-01-01/staticsites"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type StaticWebAppLinkBackendResource struct{}
+
+var _ sdk.Resource = StaticWebAppLinkBackendResource{}
+
+type StaticWebAppLinkBackendModel struct {
+	StaticWebAppID    string `tfschema:"static_web_app_id"`
+	BackendResourceId string `tfschema:"backend_resource_id"`
+}
+
+type linkedBackend struct {
+	id   string
+	name string
+}
+
+func (r StaticWebAppLinkBackendResource) ResourceType() string {
+	return "azurerm_static_web_app_link_backend"
```

Comment:
Can we call this
```suggestion
	return "azurerm_static_web_app_linked_backend"
```
as the resource names the "thing" created, not the action.